zookeeper-server/src/test/java/org/apache/zookeeper/test/BackupManagerTest.java

```diff
@@ -526,6 +527,44 @@ public void testTimetableBackup() throws Exception {
     newBackupManager.stop();
   }
 
+  @Test
+  public void testBackupFileNamingOnStreamMode() throws Exception {
+    // Test using a non-default StreamMode to see if the backup file naming is compatible with
+    // different StreamModes used in snapshot compression
+    SnapStream.StreamMode origStreamMode = SnapStream.getStreamMode();
+    SnapStream.setStreamMode(SnapStream.StreamMode.GZIP);
+
+    // Test case 1: make sure backup files are generated correctly with StreamMode extension
+    backupManager.initialize();
+    createNodes(connection, 110);
+    backupManager.getSnapBackup().run(1);
+    String[] expectedBackupFiles = getBackupFiles(backupDir);
+
+    // Sort the snapshot backup file name array because the first snapshot may not have
+    // the stream mode set. We want to look at the latest snapshot file and check if it contains
+    // the stream mode extension
+    Arrays.sort(expectedBackupFiles);
+    String newestSnapshotBackupFileName = expectedBackupFiles[expectedBackupFiles.length - 1];
+    Assert.assertTrue(
+        newestSnapshotBackupFileName.endsWith(SnapStream.StreamMode.GZIP.getFileExtension()));
+
+    // Test case 2: when the backup file is read into a BackupFileInfo object (for restoration),
+    // make sure the standard file name is created correctly in the following format:
+    // snapshot.<starting zxid>.<stream mode extension>
+    File[] snapshotBackups = backupDir.listFiles(
+        f -> f.getName().startsWith(Util.SNAP_PREFIX) && f.getName()
+            .endsWith(SnapStream.StreamMode.GZIP.getFileExtension()));
+    Assert.assertNotNull(snapshotBackups);
+
+    BackupFileInfo compressedSnapshotBackupFileInfo =
+        new BackupFileInfo(snapshotBackups[0], BackupFileInfo.NOT_SET, BackupFileInfo.NOT_SET);
+    String standardFileName = compressedSnapshotBackupFileInfo.getStandardFile().getName();
+    Assert.assertTrue(standardFileName.endsWith(SnapStream.StreamMode.GZIP.getFileExtension()));
+
```

Comment:
Could you test if the change works on copyToLocalStorage() as well?

Reply:
I also ran the test `FileSystemBackupStorageTest.java` that tests `copyToLocalStorage()` and the test passes without issues.

Conceptually, these methods are responsible for moving/streaming files as a whole and therefore do not do any parsing work of the file name itself (it plays around with paths but these are delimited by '/'), so file names are kept across these operations. So I am not too concerned that this change will affect these methods.

Are there any specific things we should look out for in case I missed anything?